api/app/signals/apps/services/domain/filescanner.py

```diff
@@ -0,0 +1,44 @@
+# SPDX-License-Identifier: MPL-2.0
+# Copyright (C) 2021 Gemeente Amsterdam
+"""
+Check file types, both for fresh uploads and stored files.
+"""
+import mimetypes
+
+import magic
+
+ALLOWED_MIME_TYPES = ['image/jpeg', 'image/png', 'image/gif']
+
+
+class BadFileError(Exception):
+    pass
+
+
+class UploadScannerService:
+    def _get_mime_from_extension(uploaded_file):
+        return mimetypes.guess_type(uploaded_file.name)[0]  # not interested in encoding
+
+    @staticmethod
+    def _get_mime_from_contents(uploaded_file):
+        for chunk in uploaded_file.chunks(chunk_size=2048):
+            return magic.from_buffer(chunk, mime=True)  # only interested in the start of a file
+
+    def scan_file(uploaded_file):
```

Comment:
nitpick but if `_get_mime_from_contents` is a `staticmethod` then this should also be a `staticmethod`